build-release-tools/application/nightly_build_clean_up.py

```diff
@@ -0,0 +1,240 @@
+                        action='store')
+
+    parser.add_argument('--bintray-subject',
+                        help="The target subject you want to clean. Default: rackhd",
+                        action='store')
+
+    parser.add_argument('--bintray-repo',
+                        help="Repo of the subject you want to clean. Default: debian",
+                        action='store')
+
+    parser.add_argument('--dockerhub-cred',
+                        help="user:pass or just a ***** token.",
+                        action='store')
+
+    parser.add_argument('--dockerhub-repo',
+                        help="Images to be cleaned in which Repo.",
+                        action='store')
+
+    parser.add_argument('--dockerhub-api-url',
+                        help="Base API URL for dockerhub.",
+                        action='store')
+
+    parser.add_argument('--atlas-cred',
+                        help="user:token of atlas.",
+                        action='store')
+
+    parser.add_argument('--atlas-url',
+                        help="Base URL for Atlas, default: https://atlas.hashicorp.com/api/v1",
+                        action='store')
+
+    parser.add_argument('--atlas-name',
+                        help="The repo name under a specific account of atlas, default: rackhd",
+                        action='store')
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+RACKHD_REPOS = ["on-imagebuilder", "on-core", "on-syslog", "on-dhcp-proxy", "files", \
+                "on-tftp", "on-wss", "on-statsd", "on-tasks", "on-taskgraph", "on-http"]
+
+def clean_bintray_nightly_builds(bintray_cred, bintray_subject, bintray_repo, date_interval):
+    """
+    clean bintray nightly builds
+    """
+    bintray = Bintray(creds=bintray_cred, subject=bintray_subject, repo=bintray_repo)
+    for package in RACKHD_REPOS:
+        versions = bintray.get_package_versions(package)
+        for version in versions:
+            build_date = get_build_date_from_version(package, version)
+            if not build_date:
+                continue
+            if date_interval[0] <= build_date <= date_interval[1]:
+                print "Cleaning {0}/{1} in bintray".format(package, version)
+                bintray.del_package_version(package, version)
+    print "Bintray nightly builds between {0} and {1} has been cleaned successfully".format(date_interval[0], date_interval[1])
+
+def clean_dockerhub_nightly_builds(dockerhub_cred, dockerhub_repo, dockerhub_api_url, date_interval):
+    """
+    clean dockerhub nightly builds
+    """
+    dockerhub = Dockerhub(creds=dockerhub_cred, repo=dockerhub_repo, api_url=dockerhub_api_url)
+    for package in RACKHD_REPOS:
+        tags = dockerhub.get_package_tags(package)
+        # for rackhd docker images, version==tag
+        for tag in tags:
+            build_date = get_build_date_from_version(package, tag)
+            if not build_date:
```

Comment:
comment please "for release build versioned X.Y.Z , the build_date will be NULL"